draft-add-requirements.md

```diff
@@ -142,18 +243,100 @@ Narrative required.
+As stated above, an enterprise-managed Wi-Fi network might be able to access both the Internet and a private
+intranet. The private domains managed by the enterprise might only be resolvable using a specific DNS resolver,
+hence use of that resolver is essential for such domains. However it does not necessarily mean that all queries
+for all domains have to flow through that resolver.
+
+Only sending the necessary queries through the enterprise resolver, and not generic Internet queries, has the privacy
+benefit of only exposing traffic to the enterprise that fall within a limited set of domains.
+
+Using encrypted DNS for private names also opens up the possibility of doing private name resolution outside
+of the content of a VPN or managed network. If the DNS resolver authenticates clients, it can offer its resolver
+for private names on a publicly accessible server, while still limiting the visibility of the DNS traffic.
 
 | Requirement | Description |
 | R11.1 | Example requirement |
 
-## Discover a mapping between a public domain and a resolver {#last-usecase}
+{: #last-usecase}
+## Encrypted resolvers for content providers {#cdn-content}
 
-Narrative required.
+Content Delivery Networks (CDNs), and content-providers more broadly, can also provide encrypted DNS resolvers that can
+be used by clients over the public Internet. These resolvers can either allow resolution of all public names (like
+normal recursive resolvers), or be designed to serve a subset of names managed by the content provider (like an
+authoritative resolver). Using these resolvers can allow the content provider to directly control how DNS answers are
+used for load balancing and address selection, which could improve performance of connections to the content provider.
+
+Using a content-provider's encrypted resolver can also provide several privacy and security benefits: 
+
+- Prevent devices along the network path to the content-provider's resolver from observing client DNS messages
+- Verify that answers come from the entity that manages the domains being resolved
+- Reduce the number of entities able to monitor the specific names accessed by a client to only the client and the content provider, assuming that the content provider would already see the names upon a secure connection later being made based on the DNS answers (e.g., in the TLS SNI extension)
+
+However doing so can create other issues. (To be described)
 
 | Requirement | Description |
 | R12.1 | Example requirement |
 
+# Privacy and security requirements {#priv-sec}
+
+Encrypted DNS improves the privacy and security of DNS queries and answers in the presence of malicious
+attackers. Such attackers are assumed to interfere with or otherwise impede DNS traffic and corresponding
+discovery mechanisms. They may be on-path or off-path between the client and entities with which the client
+communicates {{?RFC3552}}. These attackers can inject, tamper, or otherwise interfere with traffic as needed.
```

Comment:
On the thread, we discussed limiting the capabilities of this attacker to not worry about proving that a local network doesn't have an encrypted resolver, etc. Chris Wood, can you help here?

Reply:
An on-path attacker typically will not be always present in the network; the client can remember that a network-provided encrypted resolver is provided by the network (e.g., store the certificate fingerprint or authentication domain name associated with SSID:BSSID). If the endpoint cannot discover the network-provided resolver when it re-attaches to the network, it can assume an attacker blocked the discovery.